daemon/info.go

```diff
@@ -111,3 +112,12 @@ func (daemon *Daemon) SystemInfo() (*types.Info, error) {
 
 	return v, nil
 }
+
+// HTTP_PROXY, HTTPS_RPOXY and NO_PROXY are case-insensitive.
```

Comment:
Not quite accurate. They can be either all upper, or all lower. `HTTP_proxy` won't work for example.